860 activity task ssantiago (#1101)
temporalio/documentation: docs/concepts/what-is-an-activity-task-execution.md

  - explanation
---

> An Activity Task Execution results from the invocation of Activity Function Definition.

Reviewer: I wouldn't say it "results from".
An Activity Task Execution is when the Worker uses the context provided from the Activity Task and executes the Activity Function Definition.

Another way to look at it - is it is the period between the ActivityTaskStarted and ActivityTaskCompleted (or other Closed) Events.

  - explanation
---

An Activity Task Execution is when the Worker uses the Context provided from the [Activity Task](/docs/concepts/what-is-an-activity-task) and executes the [Activity Definition](/docs/concepts/what-is-an-activity-definition) (also known as the Activity Function).

> The [ActivityTaskScheduled Event](/docs/references/events/activitytaskscheduled) corresponds to when the Temporal Cluster puts the Activity Task into the Task Queue.

Reviewer: ```suggestion
The [ActivityTaskScheduled Event](/docs/references/events#activitytaskscheduled) corresponds to when the Temporal Cluster puts the Activity Task into the Task Queue.
```

I apologize - I gave you bad links which is causing the build to fail.

  - explanation
---

An Activity Task Execution is when the Worker uses the Context provided from the [Activity Task](/docs/concepts/what-is-an-activity-task) and executes the [Activity Definition](/docs/concepts/what-is-an-activity-definition).

> From the API, users see an Activity being executed and assume that it runs once.
> 
> In reality, the Activity can be retried per a Retry Policy until it achieves its expected result.

Reviewer: ```suggestion
The API to schedule an Activity Execution provides an "effectively once" experience, even though there may be several Activity Task Executions that take place to successfully complete an Activity.
```

Reviewer: Could add more about the Events associated with an Activity Task Execution - like I mention below.

> The Temporal Cluster puts an Activity Task into the Task Queue. This correlates to the [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks) in the Activity Task's Context.

Reviewer: ```suggestion
The [ActivityTaskScheduled Event](/docs/references/events/activitytaskscheduled) corresponds to when the Temporal Cluster puts the Activity Task into the Task Queue.

The [ActivityTaskStarted Event](/docs/references/events/#activitytaskstarted) corresponds to when the Worker picks up the Activity Task from the Task Queue.

Either [ActivityTaskCompleted](/docs/references/events/activitytaskcompleted) or one of the other Closed Activity Task Events corresponds to when the Worker has yielded back to the Temporal Cluster.
```
